### PR TITLE
Render compact boundaries in task threads

### PR DESCRIPTION
**Pull request by lsm**
Render compact boundary events as distinct yellow task-thread rows with trigger and token metadata.

Tests:
- bun run check
- cd packages/web && bunx vitest run src/components/space/thread/minimal/MinimalThreadFeed.test.tsx

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

**Recommendation: REQUEST_CHANGES**

### Summary

Two distinct changes bundled in one PR. The compact boundary UI rendering (task scope) is clean and well-implemented. The compaction trigger refactor for non-native providers is unrelated to the task and introduces a behavioral regression.

### Findings

**P1 — Scope creep: compaction refactor unrelated to task**
The task asks for compact boundary rendering in the thread view. The PR also includes a significant refactor of auto-compaction behavior for non-native providers (4 daemon files, ~55 lines changed). This changes how compaction works in production for all non-Anthropic providers. The commit message `feat: render compact boundaries in task threads` does not reflect this. The compaction refactor should be a separate PR with its own task, description, and review.

Changed files outside task scope:
- `packages/daemon/src/lib/agent/query-options-builder.ts`
- `packages/daemon/src/lib/agent/sdk-message-handler.ts`
- `packages/daemon/tests/unit/1-core/agent/query-options-builder.test.ts`
- `packages/daemon/tests/unit/1-core/agent/sdk-message-handler.test.ts`

**P1 — Compaction regression for unknown-context-window models**
When `contextWindow` is unavailable for a non-native provider, the new code sets `{ autoCompactEnabled: false }` with no fallback. The old code had a NeoKai-level trigger in `SDKMessageHandler` that would manually enqueue `/compact` at 85% threshold. Now both layers are disabled — context grows unbounded until the upstream API rejects the request. For any non-native provider model missing from the models cache, this is a regression. Either document this as intentional or restore the NeoKai-level fallback for unknown models.

**P2 — Dead code: `ContextTracker.shouldCompact()` and `markCompactionTriggered()` orphaned**
The compaction refactor removed the only caller of `shouldCompact()` and `markCompactionTriggered()` from `ContextTracker`. Per project guidelines: "Remove imports, variables, and functions that your changes made unused." Clean up `shouldCompact()`, `markCompactionTriggered()`, `lastCompactionTriggerAt`, `COMPACTION_THRESHOLD`, and `DEFAULT_COMPACTION_COOLDOWN_MS`.

**P3 — Test gaps for compact boundary rendering**
The compact boundary test does not assert `durationMs` rendering (test data includes `duration_ms: 2100` but no assertion checks for `"2s"`). No test for the `postTokens = undefined` fallback case. Minor but would improve coverage.

### Verdict

The compact boundary UI rendering is ready. Please split the compaction refactor into a separate PR and address the regression for unknown-context-window models.

File: packages/web/src/components/space/thread/minimal/MinimalThreadFeed.tsx [hunk 1]
```diff
@@ -749,6 +791,12 @@ function buildFeedTurns(
     };
 
     for (const row of block.rows) {
+      const compactBoundaryTurn = buildCompactBoundaryTurn(row);
+      if (compactBoundaryTurn) {
+        flushAgent();
+        turns.push(compactBoundaryTurn);
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/web/src/components/space/thread/minimal/MinimalThreadFeed.tsx:798]: Fixed in 8a02104d1. Pre-boundary flush now passes resultInfo only when pendingAgentRows contains the result row, and added a regression test for assistant → compact_boundary → assistant → result.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

**Recommendation: APPROVE**

### Re-review (commit 8a02104d1)

Previous P1 findings were false positives — daemon changes were from PR #2097 already merged to `origin/dev`; local `dev` was stale. Actual PR scope is correct: 3 web files only.

### Assessment

**space-task-thread-events.ts** — Clean. `compact_boundary` event kind with proper type guard, metadata extraction, summary formatting. `continue` correctly skips generic system handling.

**MinimalThreadFeed.tsx** — `CompactBoundaryFeedTurn` interface, builder, and component all follow existing turn-type patterns. Yellow styling consistent with reference `CompactBoundaryMessage`. Fix commit correctly scopes result metadata to the owning segment (pre-boundary flush no longer leaks result from post-boundary rows). `openSession` callback with `pushOverlayHistory` wired correctly for both node_agent and task_agent.

**MinimalThreadFeed.test.tsx** — Two tests covering: (1) compact boundary rendering with trigger, token counts, saved delta; (2) result trigger only on the segment containing the result row. Good regression coverage for the boundary-split edge case.

All CI green. All review threads resolved. Clean PR.
